tools/scripts/generic_variables.mk

```diff
@@ -0,0 +1,23 @@
+# Create variables that will be used within src.mk
+
+# This variables will be set to their default value if not set by the user
+# It assumes that the makefile if invoked from No-OS/project/some_project
+
+# User must set PLATFORM variable between:
+# PLATFORM = xilinx
+# PLATFORM = altera
+# PLATFORM = aducm3029
+
+PROJECT			?= $(realpath .)
+NO-OS			?= $(realpath ../..)
+
+INCLUDE			?= $(NO-OS)/include
+DRIVERS 		?= $(NO-OS)/drivers
+PLATFORM_DRIVERS	?= $(NO-OS)/drivers/platform/$(PLATFORM)
+
+ifeq (aducm3029,$(strip $(PLATFORM)))
+WORKSPACE		?= $(NO-OS)/projects
```

Comment:
why is this different for aducm ?

Reply:
In a workspace I have configurations that I want to use on multiple projects, that's why on aducm I used $(NO-OS)/projects . On the other hand for xilinx projects a workspace is used for each new projects (it seems that is recommended this way)